### PR TITLE
New Release Fixes

### PR DESCRIPTION
**Pull request by kingthorin**
* Incremented to 1.1.4 and changed build to keep the value synced between pom.xml and ZapAddOn.xml.
* Removed casting when loading/referencing extensions by using class not NAME string.
* Added exception handling for when spider is selected but target is not available.
* Other minor maint changes.

See zaproxy/zaproxy#5229

### CODE REVIEW

File: zaproxy/src/org/zaproxy/zap/extension/attacksurfacedetector/AttackSurfaceDetector.java [hunk 1]
```diff
@@ -83,8 +75,6 @@ public void hook(ExtensionHook extensionHook)
         }
     }
 
-    public String getMessageString(String key) { return messages.getString(key); }
```
Comment (kingthorin) [inline zaproxy/src/org/zaproxy/zap/extension/attacksurfacedetector/AttackSurfaceDetector.java:86]: This was unused. (Also likely to NPE due to messages being set null above/earlier)

File: zaproxy/src/org/zaproxy/zap/extension/attacksurfacedetector/AttackSurfaceDetector.java [hunk 2]
```diff
@@ -112,11 +102,6 @@ public URL getURL()
             return null;
         }
     }
-    @Override
-    public boolean isEnabled()
-    {
-        return true;
```
Comment (kingthorin) [inline zaproxy/src/org/zaproxy/zap/extension/attacksurfacedetector/AttackSurfaceDetector.java:118]: This is unnecessary, also problematic if the user actually disabled the addon via Options.

File: zaproxy/src/org/zaproxy/zap/extension/attacksurfacedetector/AttackSurfaceDetector.java [hunk 3]
```diff
@@ -52,13 +51,6 @@ public AttackSurfaceDetector()
         logger.debug("calling constructor");
         initialize();
         logger.debug("No-arg Constructor");
-        this.setEnabled(true);
-    }
-
-    public AttackSurfaceDetector(String name)
```
Comment (kingthorin) [inline zaproxy/src/org/zaproxy/zap/extension/attacksurfacedetector/AttackSurfaceDetector.java:58]: Never used by core.

**Comment by kingthorin** [comment]
The new warning ends up looking like this (animated gif):
asd_114

**Comment by matthewD-AVI** [comment]
Thank you for the updates, I am merging them in now and will publish a new release

**Pull request merged**